src/ert/_c_wrappers/job_queue/ext_job.py

```diff
@@ -1,263 +1,211 @@
-
-        if self.get_arglist() != other.get_arglist():
-            return False
+            path_to_executable = executable
+
+        resolved = None
+        if os.path.exists(path_to_executable):
+            resolved = path_to_executable
+        elif not os.path.isabs(executable):
+            # look for system installed executable
+            resolved = shutil.which(executable)
+
+        if resolved is None:
+            raise ConfigValidationError(
+                f"Could not find executable {executable} for job {name}"
+            )
 
-        if self.get_config_file() != other.get_config_file():
-            return False
+        if not os.access(resolved, os.X_OK):  # is not executable
+            raise ConfigValidationError(
+                f"ExtJob {name} with executable"
+                f" {resolved} does not have execute permissions"
+            )
 
-        if self.get_stderr_file() != other.get_stderr_file():
-            return False
+        if os.path.isdir(resolved):
+            raise ConfigValidationError(
+                f"ExtJob {name} has executable set to directory {resolved}"
+            )
 
-        if self.get_stdout_file() != other.get_stdout_file():
-            return False
+        return resolved
 
-        return True
+    def copy(self):
+        return replace(self)
+
+    _int_keywords = ["MAX_RUNNING", "MAX_RUNNING_MINUTES", "MIN_ARG", "MAX_ARG"]
+    _str_keywords = [
+        "STDIN",
+        "STDOUT",
+        "STDERR",
+        "START_FILE",
+        "EXECUTABLE",
+        "PORTABLE_EXE",
+        "TARGET_FILE",
+        "ERROR_FILE",
+        "START_FILE",
+    ]
+
+    @classmethod
+    def _parse_config_file(cls, config_file: str):
+        parser = ConfigParser()
+        for int_key in cls._int_keywords:
+            parser.add(int_key, value_type=ContentTypeEnum.CONFIG_INT).set_argc_minmax(
+                1, 1
+            )
+        for path_key in cls._str_keywords:
+            parser.add(path_key).set_argc_minmax(1, 1)
+
+        parser.add("EXECUTABLE", required=True).set_argc_minmax(1, 1)
+        parser.add("ENV").set_argc_minmax(1, 2)
+        parser.add("EXEC_ENV").set_argc_minmax(1, 2)
+        parser.add("DEFAULT").set_argc_minmax(2, 2)
+        parser.add("ARGLIST").set_argc_minmax(1, -1)
+        schema = parser.add("ARG_TYPE")
+        schema.set_argc_minmax(2, 2)
+        schema.iset_type(0, ContentTypeEnum.CONFIG_INT)
+
+        return parser.parse(
+            config_file,
+        )
+
+    @classmethod
+    def _read_str_keywords(cls, content_dict, config_content):
+        def set_nullable(keyword, key):
+            value = config_content.getValue(keyword)
+            if value == "null":
+                value = None
+            content_dict[key] = value
+
+        for key in cls._str_keywords:
+            if config_content.hasKey(key):
+                if key in ("STDIN", "STDOUT", "STDERR"):
+                    set_nullable(key, key.lower() + "_file")
```

Comment:
this is a change in behaviour for `STDIN` (the C code didn't check for "null") - it's not present in current logs, and probably not much used, so i don't know how much we should care.

Reply:
I guess its best to handle these consistently, even though this is potentially breaking change? The ones that would care would be those that set STDIN to null and expected there to be a file called null. Seems unrealistic. I have added it to the PR comment, title and mark, just in case. Will add it to the commit body aswell.

Reply:
treating them consistently makes sense, for sure.
i also agree that it's unrealistic that this will hit anybody : ) the logs on azure do not contain STDIN at all...